webhook/resources/backupvolume/mutator.go

```diff
@@ -45,9 +45,37 @@ func (b *backupVolumeMutator) Create(request *admission.Request, newObj runtime.
 }
 
 func (b *backupVolumeMutator) Update(request *admission.Request, oldObj runtime.Object, newObj runtime.Object) (admission.PatchOps, error) {
+	backupVolume, ok := newObj.(*longhorn.BackupVolume)
+	if !ok {
+		return nil, werror.NewInvalidError(fmt.Sprintf("%v is not a *longhorn.BackupVolume", newObj), "")
+	}
+	if err := b.addBackupsDeleteCustomResourceLabel(backupVolume); err != nil {
+		err := errors.Wrapf(err, "failed to add the label longhorn.io/delete-custom-resource-only to backups of the backup volume %v", backupVolume.Name)
+		return nil, werror.NewInvalidError(err.Error(), "")
+	}
```

Comment:
Move the label check outside of addBackupsDeleteCustomResourceLabel
```suggestion
	deleteCustomResourceOnlyLabelExists, err := datastore.IsLabelLonghornDeleteCustomResourceOnlyExisting(bv)
	if err != nil {
		return errors.Wrap(err, "failed to check if the label longhorn.io/delete-custom-resource-only exists")
	}
	
	if deleteCustomResourceOnlyLabelExists {
		if err := b.addBackupsDeleteCustomResourceLabel(backupVolume); err != nil {
			err := errors.Wrapf(err, "failed to add the label longhorn.io/delete-custom-resource-only to backups of the backup volume %v", backupVolume.Name)
			return nil, werror.NewInvalidError(err.Error(), "")
		}
	}
```

Reply:
Modified. Thanks.